engine/timer_task_queue.go

```diff
@@ -91,20 +91,27 @@ func NewTimerTaskQueueImpl(
 
 func (w *timerTaskQueueImpl) Stop(ctx context.Context) error {
 	// close timer to prevent goroutine leakage
-	w.nextPreloadTimer.Close()
-	w.nextFiringTimer.Close()
-	w.triggerPollTimer.Close()
+	w.nextPreloadTimer.Stop()
+	w.nextFiringTimer.Stop()
+	w.triggerPollTimer.Stop()
+
+	w.processor.RemoveTimerTaskQueue(w.shardId)
 
 	return nil
 }
 
 func (w *timerTaskQueueImpl) TriggerPollingTasks(req xcapi.NotifyTimerTasksRequest) {
 	if req.ShardId != w.shardId {
-		panic("shardId doesn't match")
+		w.logger.Error(fmt.Sprintf("shardId doesn't match: %d - %d", req.ShardId, w.shardId))
+		return
 	}
 	w.triggeredPollingChan <- req
 }
 
+func (w *timerTaskQueueImpl) UpdateShardId(shardId int32) {
+	w.shardId = shardId
```

Comment:
I am not entirely sure this is correct, because there are loaded tasks in memory for this queue, just updating the shardId will be broken..I think we can just shut it down and start a new one

Reply:
I think the effect is the same. Let's sync offline.

Reply:
Done.